pilopt/src/lib.rs

```diff
@@ -439,6 +440,39 @@ fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
     }
 }
 
+/// Simplifies permutations of form `0 $ [...] is S $ [...]` to `S = 0`.
+/// If the LHS sel of a permutation is 0, the LHS is the empty set.
+/// Therefore the RHS must also be the empty set.
+fn simplify_empty_permutations<T: FieldElement>(pil_file: &mut Analyzed<T>) {
+    for identity in &mut pil_file.identities.iter_mut() {
+        match identity {
+            Identity::Permutation(PermutationIdentity {
+                id,
+                source,
+                left,
+                right,
+            })
+            | Identity::PhantomPermutation(PhantomPermutationIdentity {
+                id,
+                source,
+                left,
+                right,
+            }) => {
+                if !matches!(&left.selector, AlgebraicExpression::Number(n) if n == &T::zero()) {
+                    continue;
+                }
+
+                let pol = PolynomialIdentity {
+                    id: *id,
+                    expression: right.selector.clone(),
+                    source: source.clone(),
+                };
+                *identity = Identity::Polynomial(pol);
```

Comment:
Can we just remove the phantom? Shouldn't we keep it so that the multiplicity can be set to zero?

Reply:
Ah, there is no multiplicity. Hm.

Reply:
I'm not too aware of that stuff, but if changing the multiplicity only makes more sense can do that too

Reply:
What I'm saying is: If we remove the phantom, we should also remove the columns and constraints generated from the phantom. We had a similar problem before, but not sure how we resolved it.

Reply:
Do you have a simple example of the other constraints that are generated?
I think it could be the case that they are still sound regardless, because we're not removing the phantom permutation, rather making them unsatisfiable.